src/thor/bidirectional_astar.cc

```diff
@@ -837,8 +837,11 @@ bool BidirectionalAStar::SetForwardConnection(GraphReader& graphreader, const BD
   } else {
     // If no predecessor on the forward path get the predecessor on
     // the reverse path to form the cost.
-    uint32_t predidx = opp_pred.predecessor();
-    float oppcost = (predidx == kInvalidLabel) ? 0 : edgelabels_reverse_[predidx].cost().cost;
+    uint32_t opp_predidx = opp_pred.predecessor();
+    // TODO(nils): this can't be happen, it'd mean that this is a trivial route
+    // , which is not supposed to be here! same in SetReverseConnection
+    float oppcost =
+        (opp_predidx == kInvalidLabel) ? 0.f : edgelabels_reverse_[opp_predidx].cost().cost;
```

Comment:
this check shouldn't be necessary, it'd mean we have a trivial route in bidir a*